astronomy/date_test.cpp

```diff
@@ -125,6 +139,122 @@ TEST_F(DateTest, LeapSecond) {
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1963-10-31T24:00:00,000"_UTC - 0.100 * Second,
+                            "1963-10-31T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1963-11-01T00:00:00"_UTC, "1963-11-01T00:00:02,697"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1964-03-31T24:00:00,000"_UTC - 0.100 * Second,
+                            "1964-03-31T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1964-04-01T00:00:00"_UTC, "1964-04-01T00:00:02,984"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1964-08-31T24:00:00,000"_UTC - 0.100 * Second,
+                            "1964-08-31T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1964-09-01T00:00:00"_UTC, "1964-09-01T00:00:03,282"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1964-12-31T24:00:00,000"_UTC - 0.100 * Second,
+                            "1964-12-31T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1965-01-01T00:00:00"_UTC, "1965-01-01T00:00:03,540"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1965-02-28T24:00:00,000"_UTC - 0.100 * Second,
+                            "1965-02-28T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1965-03-01T00:00:00"_UTC, "1965-03-01T00:00:03,717"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1965-06-30T24:00:00,000"_UTC - 0.100 * Second,
+                            "1965-06-30T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1965-07-01T00:00:00"_UTC, "1965-07-01T00:00:03,975"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1965-08-31T24:00:00,000"_UTC - 0.100 * Second,
+                            "1965-08-31T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1965-09-01T00:00:00"_UTC, "1965-09-01T00:00:04,155"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1968-01-31T24:00:00,000"_UTC - 0.100 * Second,
+                            "1968-01-31T23:59:59,800"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1968-02-01T00:00:00"_UTC, "1968-02-01T00:00:06,186"_TAI),
+      Lt(0.5 * Milli(Second)));
+
+  EXPECT_THAT(AbsoluteError("1971-12-31T24:00:00,000"_UTC - 0.107'7580 * Second,
+                            "1971-12-31T23:59:60,000"_UTC),
+              Lt(1 * Micro(Second)));
+  EXPECT_THAT(
+      AbsoluteError("1972-01-01T00:00:00"_UTC, "1972-01-01T00:00:10,000"_TAI),
+      Lt(0.5 * Milli(Second)));
+}
+
+TEST_F(DateTest, Pre1972Rates) {
+  // Check that cancellations aren't destroying the test.
+  EXPECT_THAT("1961-01-01T00:00:00"_UTC + 1 * Minute / (1 - 150e-10),
```

Comment:
Why `EXPECT_THAT(Ne)` and not `EXPECT_NE`?  Is this to avoid fuzzy equality tests for reals?  If so, comment.